source/jormungandr/jormungandr/api.py

```diff
@@ -41,10 +41,18 @@
 import logging
 from jormungandr.new_relic import record_custom_parameter
 from jormungandr.authentication import get_user, get_token, get_app_name, get_used_coverages
-# http://www.gevent.org/intro.html#monkey-patching
-#import gevent.monkey
-#gevent.monkey.patch_socket()
-#gevent.monkey.patch_ssl()
+
+if rest_api.app.config.get('PATCH_WITH_GEVENT_SOCKET', False):
+    import gevent.socket
+    import urllib3
+    logger = logging.getLogger('jormungandr.patch_gevent_socket')
+    logger.info("Attention! You'are patching urllib3.connection.connection.socket with gevent.socket, "
+                "parallel calling valhalla is activated")
+    # This line replaces the gevent.monkey.patch_socket()
+    # the reason why we don't use patch_socket() at the very beginning of jormungandr is
+    # that it caused a mysterious performance regression for certain requests, thus we patch
+    # only at places where asynchronisation is needed
```

Comment:
Ok @kinnou02 explained this to me.

maybe it would be good to have a bit more precise explanation in this comment (which is already nice btw!) explaining that we monkey_patch only http request because we want asynchronisation on those, but we don't want to activate it for reddis because it causes performance regression